tests/lib/heap/testcase.yaml

```diff
@@ -13,7 +13,7 @@ tests:
       - qemu_xtensa
       - esp32s2_saola
       - esp32s2_lolin_mini
-    filter: not CONFIG_SOC_NSIM
+    filter: not SOC_FAMILY_NSIM_ARC_CLASSIC
```

Comment:
Not in scope for this PR, but what's the reason this test is disabled on nSIM?  The heap code is a pretty core thing, and you're including it via minimal/picolibc already; surely we want it tested...

Reply:
Hi @andyross,
Please take a look at https://github.com/zephyrproject-rtos/zephyr/pull/73892.

Reply:
We remove this filter in https://github.com/zephyrproject-rtos/zephyr/pull/73892 and allow test for all nSIM platforms now.